src/dune_lang/syntax.ml

```diff
@@ -203,27 +203,40 @@ module Error = struct
           ]
       :: repl )
 
-  let disabled loc t ~dune_lang_ver ~what =
-    let min_lang_version, min_dune_version =
-      let major, major_map =
-        Option.value_exn (Int.Map.min_binding t.supported_versions)
-      in
-      let minor, lang = Option.value_exn (Int.Map.min_binding major_map) in
-      ((major, minor), lang)
-    in
+  let inactive loc t ~dune_lang_ver ~what =
     User_error.raise ~loc
-      [ Pp.textf
-          "%s is available only when %s is enabled in the dune-project file. \
-           It cannot be enabled automatically because the currently selected \
-           version of dune (%s) does not support this plugin.\n\
-           You must enable it using (using %s ..) in your dune-project file. \
-           The first version of this plugin %s was introduced in dune %s."
-          what t.name
-          (Version.to_string dune_lang_ver)
-          t.name
-          (Version.to_string min_lang_version)
-          (Version.to_string min_dune_version)
-      ]
+      ( [ Pp.textf
+            "%s is available only when %s is enabled in the dune-project file. \
+             You must enable it using (using %s ..) in your dune-project file."
+            what t.name t.name
+        ]
+      @
+      if t.experimental then
+        []
+      else
+        match
+          Supported_versions.greatest_supported_version_for_dune_lang
+            ~dune_lang_ver t.supported_versions
+        with
+        | None ->
+          let min_lang_version, min_dune_version =
+            let major, major_map =
+              Option.value_exn (Int.Map.min_binding t.supported_versions)
+            in
+            let minor, lang =
+              Option.value_exn (Int.Map.min_binding major_map)
+            in
+            ((major, minor), lang)
+          in
+          [ Pp.textf
+              "Note however that the currently selected version of dune (%s) \
+               does not support this plugin. The first version of this plugin \
+               %s was introduced in dune %s."
+              (Version.to_string dune_lang_ver)
+              (Version.to_string min_lang_version)
+              (Version.to_string min_dune_version)
+          ]
```

Comment:
Maybe we could add a test case for this behavior ?

Reply:
Sounds good. I did that and tweaked the error message.